src/pages/steps/BookingInfoStep.tsx

```diff
@@ -0,0 +1,494 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+
+import {
+  useAddBookingInfoMutation,
+  useGetEventByIdQuery,
+} from '@/hooks/api/events';
+import { Alert, AlertVariants } from '@/ui/Alert';
+import { Button, ButtonVariants } from '@/ui/Button';
+import { DatePeriodPicker } from '@/ui/DatePeriodPicker';
+import { FormElement } from '@/ui/FormElement';
+import { Icons } from '@/ui/Icon';
+import { Inline } from '@/ui/Inline';
+import { Input } from '@/ui/Input';
+import { RadioButtonGroup } from '@/ui/RadioButtonGroup';
+import { getStackProps, Stack, StackProps } from '@/ui/Stack';
+import { Text } from '@/ui/Text';
+import { getValueFromTheme } from '@/ui/theme';
+import { Title } from '@/ui/Title';
+
+import { TabContentProps } from './AdditionalInformationStep';
+
+const EMAIL_REGEX: RegExp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+const URL_REGEX: RegExp = /^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&\/=]*)$/;
+const PHONE_REGEX: RegExp = /^[0-9\/_.+ ]*$/;
+
+const isValidEmail = (email: string): boolean => {
+  return !email || EMAIL_REGEX.test(email);
+};
+const isValidUrl = (url: string): boolean => {
+  return !url || URL_REGEX.test(url);
+};
+const isValidPhone = (phone: string): boolean => {
+  return !phone || PHONE_REGEX.test(phone);
+};
+
+const schema = yup
+  .object({
+    availabilityEnds: yup.string(),
+    availabilityStarts: yup.string(),
+    email: yup
+      .string()
+      .test(`email-is-not-valid`, 'email is not valid', isValidEmail),
+    phone: yup
+      .string()
+      .test(`phone-is-not-valid`, 'phone is not valid', isValidPhone),
+    url: yup
+      .string()
+      .test(`website-is-not-valid`, 'url is not valid', isValidUrl),
+    urlLabel: yup.object({
+      de: yup.string(),
+      en: yup.string(),
+      fr: yup.string(),
+      nl: yup.string(),
+    }),
+  })
+  .required();
+
+type FormData = yup.InferType<typeof schema>;
+
+const urlLabelTranslationString =
+  'create.additionalInformation.booking_info.url_type_labels';
+
+const ContactInfoType = {
+  EMAIL: 'email',
+  PHONE: 'phone',
+  URL: 'url',
+} as const;
+
+type BookingInfo = {
+  email?: string;
+  phone?: string;
+  url?: string;
+  urlLabel?: {
+    de: string;
+    en: string;
+    fr: string;
+    nl: string;
```

Comment:
```suggestion
    de?: string;
    en?: string;
    fr?: string;
    nl?: string;
```

Are these always there, so empty strings if not set? Or can they be undefined

Reply:
They should always be there with a "predefined" value.
https://documentatie.uitdatabank.be/content/json-ld-crud-api/latest/events/event-bookingInfo.html